### PR TITLE
chore: redesign dropdown in theme settings (Issue #144)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
- Issue: https://github.com/epam/ai-dial-chat/issues/144

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Deployment status: success

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Deployment status: success

**Review by denys-kolomiitsev — DISMISSED** [review]

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Deployment status: success

File: src/components/Common/ConfirmDialog.tsx [hunk 1]
```diff
@@ -26,76 +19,52 @@ export const ConfirmDialog = ({
...
-        >
-          <FloatingFocusManager
-            context={context}
-            initialFocus={confirmLabelRef}
+    <Modal
+      portalId="theme-main"
+      isOpen={isOpen}
+      onClose={() => onClose(false)}
+      dataQa="confirmation-dialog"
+      containerClassName="z-50 flex min-w-[90%] flex-col items-center gap-4 rounded bg-layer-3 p-6 text-center md:min-w-[300px] md:max-w-[500px]"
```
Comment (IlyaBondar) [inline src/components/Common/ConfirmDialog.tsx]: `rounded bg-layer-3` can be removed - already in `Modal`

File: src/components/Files/FileManagerModal.tsx [hunk 2]
```diff
@@ -323,193 +308,177 @@ export const FileManagerModal = ({
...
-              className="relative flex max-h-full flex-col gap-4 rounded bg-layer-3 md:w-[525px]"
-              ref={refs.setFloating}
-              {...getFloatingProps()}
-            >
+    <Modal
+      portalId="theme-main"
+      isOpen={isOpen}
+      onClose={() => onClose(false)}
+      dataQa="file-manager-mpdal"
+      containerClassName="relative flex max-h-full flex-col gap-4 rounded bg-layer-3 md:w-[525px]"
```
Comment (IlyaBondar) [inline src/components/Files/FileManagerModal.tsx]: 1) there was `const dismiss = useDismiss(context, { outsidePressEvent: 'mousedown' });`. I suppose we need to use `dismissProps={{ outsidePressEvent: 'mousedown' }}`. The same is in `SelectFolderModal.tsx`

2) `relative max-x-full rounded bg-layer-3` already in `Modal` an can be removed

File: src/components/Files/PreUploadModal.tsx [hunk 3]
```diff
@@ -305,143 +291,117 @@ export const PreUploadDialog = ({
...
-
-                <ErrorMessage error={errorMessage} />
-
-                <div className="flex flex-col gap-1">
-                  <div>
-                    <span className="text-xs text-secondary">
-                      {t('Upload to')}
+    <Modal
+      portalId="theme-main"
+      containerClassName="relative flex max-h-full flex-col gap-4 rounded bg-layer-3 p-6 md:min-w-[425px] md:max-w-[500px]"
```
Comment (IlyaBondar) [inline src/components/Files/PreUploadModal.tsx]: 1) previous there were `const dismiss = useDismiss(context, { outsidePressEvent: 'mousedown' });`
I suppose we should have the same i.e. `dismissProps={{ outsidePressEvent: 'mousedown' }}`. The same is in `SelectFolderModal.tsx` 

2) `relative max-h-full rounded bg-layer-3` are already in `Modal`

File: src/components/Files/SelectFolderModal.tsx [hunk 4]
```diff
@@ -188,125 +172,109 @@ export const SelectFolderModal = ({
...
-              className="relative flex max-h-full min-w-full flex-col gap-4 rounded bg-layer-3 md:min-w-[425px] md:max-w-full"
-              ref={refs.setFloating}
-              {...getFloatingProps()}
-            >
+    <Modal
+      portalId="theme-main"
+      isOpen={isOpen}
+      onClose={() => onClose(false)}
+      dataQa="select-folder-modal"
+      containerClassName="relative flex max-h-full min-w-full flex-col gap-4 rounded bg-layer-3 md:min-w-[425px] md:max-w-full"
```
Comment (IlyaBondar) [inline src/components/Files/SelectFolderModal.tsx]: `relative max-h-full rounded bg-layer-3` are already in `Modal`

File: src/components/Settings/SettingDialog.tsx [hunk 5]
```diff
@@ -78,53 +60,51 @@ export const SettingDialog: FC<Props> = ({ open, onClose }) => {
...
   return (
-    <FloatingPortal id="theme-main">
-      <div className="fixed inset-0 z-40 flex w-full items-center justify-center overflow-hidden bg-blackout p-3">
-        <div
-          ref={modalRef}
-          className="relative inline-block max-h-full w-[500px] overflow-y-auto rounded bg-layer-3 p-4 text-left align-bottom transition-all md:max-h-[400px]"
-          role="dialog"
+    <Modal
+      portalId="theme-main"
+      containerClassName="relative inline-block max-h-full w-[500px] overflow-y-auto rounded bg-layer-3 p-4 text-left align-bottom transition-all md:max-h-[400px]"
```
Comment (IlyaBondar) [inline src/components/Settings/SettingDialog.tsx]: `relative max-h-full rounded bg-layer-3 text-left` are already in `Modal`

**Review by IlyaBondar — APPROVED** [review]

**Review by denys-kolomiitsev — APPROVED** [review]

**Pull request merged**